### PR TITLE
feat: superadmin endpoint for user deletion by id

### PR DESCRIPTION
**Pull request by bruce-pain**
## Description

This PR introduces a new endpoint that allows a superadmin to soft-delete a user by their user ID. This endpoint ensures that only authenticated superadmins can perform the deletion operation, maintaining strict security and validation checks.

### Changes:

- implemented a new endpoint for user deletion
- modified user deletion logic in the user service
- wrote tests for the endpoint

## Related Issue:

based on issue #240

## Motivation and Context

- an endpoint that allows superadmin users to remove a single user's records from the database safely

## How Has This Been Tested?

The changes have been tested using pytest with the following tests:
  - `test_unauthorised_access`: Test for unauthorized access to endpoint
  - `test_successful_deletion`: Test for successful deletion of user
  - `test_non_admin_access`: Test for non admin user access to endpoint
  - `test_not_found_error`: Test for invalid user ID

### Endpoint Details

- **Route:** `DELETE /api/v1/superadmin/users/{:user_id}`
- **Method:** DELETE
- **Parameters:**
  - `user_id`: UUID string of the user to be deleted.

### Screenshots
**create a new user to be deleted**

image

**delete the newly created user by ID**

image

**deleting the same user by ID (soft-deleted user)**

image

**deleting a non existent user by ID**

image

**non super_admin user access**

image

### Requirements

- [x] Ensure an authenticated user is accessing the endpoint
- [x] Ensure that authenticated user is a superadmin
- [x] Develop server-side logic to safely soft-delete a single user
- [x] Ensure that data removal complies with security standards

### Tasks

- [x] Implement the endpoint route api/v1/superadmin/users/{:user_id}
- [x] Perform the appropriate database operations required to soft-delete a single user
- [x] Ensure that only authorized users can access the endpoint
- [x] Ensure that authorized users are also superadmins
- [x] Implement error handling for scenarios when the user is unauthorized and invalid ID, with the appropriate HTTP status codes
- [x] Ensure the endpoint returns the appropriate response body

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Goketech — APPROVED** [review]

**Comment by bruce-pain** [comment]
@zxenonx

**Review by Goketech — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Comment by bolexs** [comment]
fix merge conflict

**Comment by bruce-pain** [comment]
@bolexs merge conflict fixed

**Review by Goketech — DISMISSED** [review]

File: api/v1/routes/superadmin.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from api.v1.models.user import User
+from api.v1.services.user import user_service
+from uuid import UUID
+
+superadmin = APIRouter(prefix="/superadmin", tags=["superadmin"])
+
+
+@superadmin.delete("/users/{user_id}")
+def delete_user(
+    user_id: UUID,
```
Comment (joboy-dev) [inline api/v1/routes/superadmin.py:21]: All id should be strings

**Comment by Goketech** [comment]
@bolexs

File: api/v1/routes/superadmin.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    current_user: Annotated[User, Depends(user_service.get_current_user)],
```
Comment (joboy-dev) [inline api/v1/routes/superadmin.py:22]: There is a get_current_super_admin in the user service. This saves you the stress of checking in the routes again

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make requested changes

**Review by bolexs — DISMISSED** [review]

File: api/v1/routes/superadmin.py (hunk 2 above)
Comment (bolexs) [reply api/v1/routes/superadmin.py:22]: resolve this please

**Comment by bolexs** [comment]
resolve ur conflicts

File: api/v1/routes/superadmin.py (hunk 1 above)
Comment (bruce-pain) [reply api/v1/routes/superadmin.py]: Resolved

File: api/v1/routes/superadmin.py (hunk 2 above)
Comment (bruce-pain) [reply api/v1/routes/superadmin.py]: resolved

**Review by Goketech — DISMISSED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — DISMISSED** [review]

**Review by Goketech — APPROVED** [review]

**Comment by Goketech** [comment]
@bolexs

**Review by Goketech — DISMISSED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Comment by Goketech** [comment]
@zxenonx

**Review by zxenonx — APPROVED** [review]
Thanks @bruce-pain

**Pull request merged**